CONTRIBUTING.md

```diff
@@ -318,40 +345,37 @@ You will also need several other mkdocs plugins, like `branchcustomization` and
 To install all of this make sure you have a recent version of Python on your computer.
 The [DataLad Handbook](http://handbook.datalad.org/en/latest/intro/installation.html#python-3-all-operating-systems) provides helpful instructions for setting up Python.
 
-An easy way to install the correct version of mkdocs and all the other required extensions
-is to use the `requirements.txt` file contained in this repository,
-by using the following command:
+In general, we strongly recommend that you install all dependencies in an isolated Python environment.
+For example using `conda`, as described in the section on Python in the DataLad Handbook, linked above.
 
 ```bash
-pip install -r requirements.txt
+conda create --name bids-spec
+conda activate bids-spec
 ```
 
-However this will also install some other packages you might not want to have (like `numpy`).
-So if you only want to install what you need to build the specification,
-use the following command:
+Or alternatively using `venv`, as described in this [Real Python tutorial](https://realpython.com/python-virtual-environments-a-primer/).
+
+A short version of the commands needed to create and activate your `venv` virtual environment would look like:
 
 ```bash
-pip install \
- mkdocs \
- mkdocs-material \
- pymdown-extensions \
- mkdocs-branchcustomization-plugin \
- mkdocs-macros-plugin \
- tabulate
+python3 -m venv env
+source env/bin/activate
 ```
 
-#### 2. Download the BIDS specification [repository](https://github.com/bids-standard/bids-specification/tree/master) onto your computer
+Note that this will create a local directory called `env` within the bids-specification directory
+but that its content will not be tracked by `git` because it is listed in the `.gitignore` file.
 
-This can be done by clicking the green button on the right titled "Clone or
-download"
-or using [this link](https://github.com/bids-standard/bids-specification/archive/master.zip).
+Once you have activated your isolated Python environment,
+an easy way to install the correct version of mkdocs and all the other required extensions
+is to use the `requirements.txt` file contained in this repository as follows:
 
-#### 3. In the terminal (command line) navigate to your local version of the specification
+```bash
+pip install -U pip
+pip install -r requirements.txt
```

Comment:
```suggestion
pip install -r requirements.txt
pip install tools/schemacode/
```
Aren't we missing the `schemacode`?

Reply:
That's actually part of the `requirements.txt`, so I suspect your issue occurred because of a non-up-to-date pip and/or not calling `pip install -r requirements.txt` from the root of the repository (so that the relative link to schemacode works)

https://github.com/bids-standard/bids-specification/blob/ba11771e76978264445cfc35d52eda14d1ca30fc/requirements.txt#L7

Reply:
I've just tested it again and this line is indeed redundant.